.gitattributes

```diff
@@ -0,0 +1,92 @@
+##### Source code #####
+
+## C++ and C source files
+*.c                     text diff=cpp
```

Comment:
diff=cpp allows GitHub to visualize a diff in a 'smarter' way - using language-specific diff parser. From gitattributes man: https://git-scm.com/docs/gitattributes

- _cpp_ suitable for source code in the C and C++ languages.

i.e. _git diff --word-diff_ is easier to read with that attribute. There are several projects doing it that way. I'm not sure if it's still relevant as of 2019? But this is one example with some logical reasoning behind this practice: http://lkml.iu.edu/hypermail/linux/kernel/1108.3/01118.html

Reply:
That's cool! Is there a before/after example? I'd be surprised if GitHub has special handling for `diff=cpp` but doesn't know to apply it by default to *.cc files (considering they e.g. guess the breakdown of languages for every repo)

Reply:
Basically git itself in its own .gitattributes file prefers to explicitly specify the proper diff viewer for .c and .h files... I think we might want to keep that similar logic in our repo. Reference:
https://github.com/git/git/blob/master/.gitattributes

Maybe adding whitespace=space is also something we may want to consider, e.g.
`*.[ch] whitespace=indent,trail,space diff=cpp`
to enforce spaces instead of tabs.